salt/kafka-tool/init.sls

```diff
@@ -0,0 +1,73 @@
+#read all pillar data
+
+{% set install_dir = pillar['pnda']['homedir'] %}
+{% set packages_server = pillar['packages_server']['base_uri'] %}
+{% set release_directory = salt['pillar.get']('kafkatool:release_dir', '/opt/pnda') %}
+{% set release_version = salt['pillar.get']('kafkatool:version', '0.3.0')  %}
+{% set release_filename = 'kafka-tool-' + release_version + '.tar.gz' %}
+
+{% set p  = salt['pillar.get']('kafka', {}) %}
+{% set local_kafka_path = p.get('prefix', '/opt/pnda/kafka') %}
+{% set local_kafkat_log_path = salt['pillar.get']('kafkatool:log_dir', '/var/log/pnda/kafkatool') %}
+
+{%- set zk_ips = [] -%}
+{%- for ip in salt['pnda.kafka_zookeepers_ips']() -%}
+{%-   do zk_ips.append(ip + ':2181') -%}
+{%- endfor -%}
+
+
+kafka-tool-extract:
+  archive.extracted:
+    - name: {{ install_dir }}
+    - source: {{ packages_server }}/{{ release_filename }}
+    - source_hash: {{ packages_server }}/{{ release_filename }}.sha512.txt
+    - archive_format: tar
+    - tar_options: v
+    - if_missing: {{ release_directory }}/kafka-tool-{{ release_version }}
+
+kafka-tool-create_link:
+  file.symlink:
+    - name: {{ release_directory }}/kafka-tool
+    - target: {{ release_directory }}/kafka-tool-{{ release_version }}
+
+kafka-tool-install_packages:
+  pkg.installed:
+    - pkgs:
+      - {{ pillar['ruby-devel']['package-name'] }} 
+      - {{ pillar['g++']['package-name'] }}
+      - {{ pillar['patch']['package-name'] }}
+
+{% if grains['os'] == 'Ubuntu' %}
+kafka-tool-install_build_essential:
+  pkg.installed:
+    - name: build-essential
+{% endif %}
+
+# gem install kafakt-0.3.0
```

Comment:
remove the comment, wrong and not needed

Reply:
removed